### PR TITLE
CI revisions re. deployment

### PR DESCRIPTION
**Pull request by maxkfranz**
**General information**

Associated issues: #152 

**Checklist**

Author:

- [x] One or more reviewers have been assigned.
- [x] Automated tests have been included in this pull request, if possible, for the new feature(s) or bug fix. -- Already there with green CI tests & Caprover
- [x] The associated GitHub issues are included (above).
- [x] Notes have been included (below).

Reviewers:

- [x] All automated checks are passing (green check next to latest commit).
- [x] At least one reviewer has signed off on the pull request.  Reviewers have two business days to review the pull request, after which the author may merge in the pull request unilaterally.

**Notes**

- Revise the docker config to better support Puppeteer and history snapshots (cytosnap).
- Use `.env.defaults` since it works better with a dockerfile in the root.
- Unfortunately, Caprover doesn't run the tests well.  GitHub does, though.  It may not make sense to run the test in the Caprover servers anyway, since the tests edit the referenced DB.  We don't want to delete networks on the test servers while people are using them.

### CODE REVIEW

**Comment by maxkfranz** [comment]
Current WIP

**Comment by maxkfranz** [comment]
@d2fong @mikekucera 

The merge conflicts are resolved.  This is ready for a once-over review so that it can be merged in

File: src/server/routes/api/thumbnail.js
Comment (d2fong) [inline src/server/routes/api/thumbnail.js:15]: So the idea here is to keep one instance of cytosnap and use the same instance everytime `runCytosnap` is called?

Comment (maxkfranz) [reply src/server/routes/api/thumbnail.js:15]: Yes.  Each cytosnap instance has an instance of chrome, which is heavy

**Comment by d2fong** [comment]
Ok looks good to me.  I think Mike should take a look as well though.

**Review by d2fong — APPROVED** [review]

**Review by mikekucera — APPROVED** [review]

**Pull request merged**
